Kconfig.nrf

```diff
@@ -64,6 +64,13 @@ config INIT_ARCH_HW_AT_BOOT
 	  leave the system in a clean state, so it is necessary to perform
 	  architecture specific hardware initialization.
 
+DT_COMPAT_JEDEC_SPI_NOR := jedec,spi-nor
+config SPI_NOR
+	default y if $(dt_compat_on_bus,$(DT_COMPAT_JEDEC_SPI_NOR),spi) && PM_EXTERNAL_FLASH_MCUBOOT_SECONDARY
+
+config SPI
+	default y if $(dt_compat_on_bus,$(DT_COMPAT_JEDEC_SPI_NOR),spi)
```

Comment:
Both settings are related to SPI NOR flashes, so are the kconfig setting names `SPI` and `SPI_NOR` really the best names available to choose between the two cases ?

Just looking at those kconfig setting names, I wouldn't be able to remember which one is used for which case.

Reply:
let me try to see if I understand you correctly.

What you would like to indicate, is that [SPI](https://github.com/nrfconnect/sdk-zephyr/blob/9271a81c5d32b3c6159cde782568399121b5cd16/drivers/spi/Kconfig#L9-L10) is default enabled when you have a nor flash ([SPI_NOR](https://github.com/nrfconnect/sdk-zephyr/blob/9271a81c5d32b3c6159cde782568399121b5cd16/drivers/flash/Kconfig.nor#L4-L5) )connected on SPI, according to devicetree ?


Maybe it's not this PR, but upstream that `SPI_NOR` is a bit too short in setting name.

For this PR, some good comments on what this Kconfig code does would be helpful.

Reply:
`What you would like to indicate, is that SPI is default enabled when you have a nor flash (SPI_NOR )connected on SPI, according to devicetree ?` yes.

Reply:
Some good comment will be highly appreciated then.
Code wise, this looks correct.